packages/amplication-git-pull-service/src/core/gitPullEvent/gitPullEvent.service.ts

```diff
@@ -1,7 +1,190 @@
+      );
+
+      const previousReadyCommit = await this.findPreviousReadyCommit(
+        pushEventMessage
+      );
+
+      if (!previousReadyCommit) {
+        await this.cloneRepository(
+          provider as GitProviderEnum,
+          pushEventMessage,
+          currentCommitDir
+        );
+      } else {
+        await this.managePullEventStorage(
+          pushEventMessage,
+          previousReadyCommit.commit,
+          repositoryDir,
+          currentCommitDir
+        );
+
+        if (this.skipPrismaValue !== 0) {
+          await this.manageDelete(
+            previousReadyCommit.id,
+            `${repositoryDir}/${previousReadyCommit.commit}`
+          );
+        }
+      }
+
+      await this.gitPullEventRepository.update(
+        newPullEventRecord.id,
+        EnumGitPullEventStatus.Ready
+      );
+    } catch (err: any) {
+      this.logger.error(
+        GitPullEventService.name,
+        { err: err.message },
+        LoggerMessages.error.CATCH_ERROR_MESSAGE,
+        "pushEventHandler method"
+      );
+    }
+  }
+
+  private async createPullEventRecordOnDB(
+    pushEventMessage: PushEventMessage
+  ): Promise<{ id: bigint }> {
+    const { installationId, ...gitPullEventParams } = pushEventMessage;
+    const newPullEventRecord = await this.gitPullEventRepository.create({
+      ...gitPullEventParams,
+      status: EnumGitPullEventStatus.Created,
+    });
+
+    this.logger.log(
+      LoggerMessages.log.NEW_REPOSITORY_WAS_CREATED,
+      GitPullEventService.name,
+      { newPullEventRecord }
+    );
+
+    return newPullEventRecord;
+  }
+
+  private async findPreviousReadyCommit(
+    pushEventMessage: PushEventMessage
+  ): Promise<EventData | undefined> {
+    const { provider, repositoryOwner, repositoryName, branch, pushedAt } =
+      pushEventMessage;
+    const previousReadyCommit =
+      await this.gitPullEventRepository.findByPreviousReadyCommit(
+        provider,
+        repositoryOwner,
+        repositoryName,
+        branch,
+        pushedAt,
+        this.skipPrismaValue
+      );
+
+    this.logger.log(
+      LoggerMessages.log.FOUND_PREVIOUS_READY_COMMIT,
+      GitPullEventService.name,
+      {
+        previousReadyCommit,
+      }
+    );
+
+    return previousReadyCommit;
+  }
+
+  private async cloneRepository(
+    provider: GitProviderEnum,
+    pushEventMessage: PushEventMessage,
+    baseDir: string
+  ): Promise<void> {
+    const accessToken = await this.gitHostProviderFactory
+      .getHostProvider(provider as GitProviderEnum)
+      .createInstallationAccessToken(pushEventMessage.installationId);
+
+    this.logger.log(
+      LoggerMessages.log.GENERATE_OCTOKIT_ACCESS_TOKEN,
+      GitPullEventService.name,
+      {
+        accessToken,
+      }
+    );
+
+    await this.gitClientService.clone(pushEventMessage, baseDir, accessToken);
+
+    this.logger.log(LoggerMessages.log.CLONE_SUCCESS, GitPullEventService.name);
+  }
+
+  private async managePullEventStorage(
```

Comment:
managePullEventStorage is suggesting that this function does something with the storage - what it does with the storage is unclear....

this function brakes the single point of responsibility as it does copy and pull.

please rename the function for pull changes from existing or something that suggest the actual function role. you may think of a different solution as well.